django/db/models/sql/compiler.py

```diff
@@ -408,9 +408,8 @@ def get_combinator_sql(self, combinator, all):
                 # If the columns list is limited, then all combined queries
                 # must have the same columns list. Set the selects defined on
                 # the query on all combined queries, if not already set.
-                if (not compiler.query.values_select and not compiler.query.annotations and
-                        self.query.values_select):
-                    compiler.query.set_values(self.query.values_select)
+                if not compiler.query.values_select and self.query.values_select:
+                    compiler.query.set_values((*self.query.values_select, *self.query.annotation_select))
```

Comment:
Do you want `set_values(*self.query.values_select, *self.query.annotation_select)` instead?

Reply:
Ahh nvm, the function accepts a tuple and not `*args`.